probePluginBazel/src/main/scala/com/virtuslab/intellij/bazel/BazelProbeHandlerContributor.scala

```diff
@@ -0,0 +1,30 @@
+package com.virtuslab.intellij.bazel
+
+import com.intellij.openapi.actionSystem.{ActionManager, AnActionEvent, DataContext}
+import com.intellij.openapi.application.ApplicationManager
+import com.virtuslab.bazel.protocol.BazelEndpoints
+import org.virtuslab.ideprobe.ProbeHandlerContributor
+import org.virtuslab.ideprobe.ProbeHandlers.ProbeHandler
+
+import com.google.idea.blaze.base.settings.BlazeUserSettings
+
+class BazelProbeHandlerContributor extends ProbeHandlerContributor {
+  override def registerHandlers(handler: ProbeHandler): ProbeHandler = {
+    handler
+      .on(BazelEndpoints.SetupBazelExecutable)(setupBazelExecutable)
+  }
+
+  def setupBazelExecutable(setup: String): Unit = {
+    ApplicationManager.getApplication.invokeAndWait(
```

Comment:
you could extend `IntellijApi` trait and use helpers from there. AFAIR: `invokeAndWait { write { BazelUserSettings.getInstance.setBaselBinaryPath(setup) } }`